sources/packages/backend/apps/queue-consumers/test/helpers/mock-utils/cas-response.factory.ts

```diff
@@ -95,3 +103,44 @@ export function createFakeCASCreateSupplierAndSiteResponse(): CreateSupplierAndS
     },
   };
 }
+
+/**
+ * Create a fake CreateSupplierNoSite response with missing postal code.
+ * @returns fake CreateSupplierNoSite response.
+ */
+export function createFakeCASCreateSupplierNoSiteResponse(options?: {
+  initialValues: {
+    supplierNumber: string;
+    supplierAddress: SupplierAddress;
+  };
+}): CreateSupplierAndSiteResponse {
+  const supplierNumber =
+    options?.initialValues?.supplierNumber ??
+    faker.datatype.number({ min: 1000000, max: 9999999 }).toString();
+  return {
+    submittedData: {
+      SupplierNumber: supplierNumber,
+      SupplierAddress: [
+        {
+          AddressLine1:
+            options?.initialValues?.supplierAddress.addressLine1 ??
+            faker.address.streetAddress(false).toUpperCase(),
+          AddressLine2: "",
+          AddressLine3: "",
+          City: options?.initialValues?.supplierAddress.city ?? "Victoria",
+          Province:
+            options?.initialValues?.supplierAddress.provinceState ?? "BC",
+          Country: options?.initialValues?.supplierAddress.country ?? "CA",
+          PostalCode:
+            options?.initialValues?.supplierAddress.postalCode ?? "H1H1H1",
+          EmailAddress: faker.internet.email(),
+        },
+      ],
+    },
+    response: {
+      supplierNumber: supplierNumber,
+      supplierSiteCode:
+        options?.initialValues?.supplierAddress.supplierSiteCode ?? "001",
```

Comment:
minor. similar to supplier number at line 117, site code can be declared as const.